src/command_modules/azure-cli-iot/azure/cli/command_modules/iot/tests/latest/test_iot_pnp_commands.py

```diff
@@ -0,0 +1,102 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+# pylint: disable=too-many-statements
+
+from azure.cli.testsdk import ScenarioTest
+
+
+class IoTPnPTest(ScenarioTest):
+
+    def test_iot_pnp(self):
+        repo_endpoint = 'https://provider.azureiotrepository-test.com'
+        repo_name = 'clitestrepo'
+        repo_name2 = 'clitestrepo-updated'
+        user_role = 'Reader'
+        user_role2 = 'Admin'
+
+        # Test 'az iot pnp repository create'
+        repo = self.cmd('iot pnp repository create -e "{0}" -n {1}'.format(repo_endpoint, repo_name),
```

Comment:
FYI, you can use self.kwargs to store any constants you want to use within calls to self.cmd()
```suggestion
        repo = self.cmd('iot pnp repository create -e "{endpoint}" -n {name}',
```

Reply:
Updated